src/core/qgsvectorlayer.h

```diff
@@ -1992,6 +2018,43 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     bool isEditCommandActive() const { return mEditCommandActive; }
 
+    /**
+     * If the `removeDuplicateNodes` property is set on a layer, whenever a new feature enters
+     * the edit buffer or the geometry of an existing feature is changed, duplicate nodes will
+     * automatically be removed without any user intervention.
+     *
+     * \since QGIS 3.4
+     */
+    bool removeDuplicateNodes() const;
+
+    /**
+     * If the `removeDuplicateNodes` property is set on a layer, whenever a new feature enters
+     * the edit buffer or the geometry of an existing feature is changed, duplicate nodes will
+     * automatically be removed without any user intervention.
+     *
+     * \since QGIS 3.4
+     */
+    void setRemoveDuplicateNodes( bool removeDuplicateNodes );
```

Comment:
Why not just expose these settings through a reference to GeometryOptions? The vector layer API is already very cluttered, and I'm expecting more options will be added to GeometryOptions in future which would mean even more API additions here...

Reply:
Can do.  I was 50/50 on that before. Clutter is not nice, but loosing time for not finding methods is also not nice... I'm not sure what's the best thing here in the long run.